src/Components/JoinExistingList.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from 'react';
+import { db } from '../lib/firebase';
+import { useHistory } from 'react-router-dom';
+
+export default function JoinExistingList({ setToken }) {
+  const [input, setInput] = useState('');
+  let history = useHistory();
+
+  const handleClick = (event) => {
+    event.preventDefault();
+    if (input === '') {
+      alert('Enter a valid token');
+    } else {
+      db.collection('lists')
+        .doc(input)
+        .onSnapshot(function (querySnapshot) {
+          if (querySnapshot.exists) {
```

Comment:
Also not a blocker, but consider making this conditional case-insensitive so that the token they type doesn't need to be lowercase to be considered a match, which could be good for usability. A few use cases come to mind, such as if the user is on their phone with auto-capitalization turned on, or if they have caps lock turned on.